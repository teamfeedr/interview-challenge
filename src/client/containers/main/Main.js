import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { usePost } from '../../utils/hooks';
import Header from '../../components/header/Header';
import Input from '../../components/input/Input';
import List from '../../components/list/List';
import { ErrorText, Title } from '../../components/text/Text';

export default function Main() {
  const [input, setInput] = useState('');
  const [previewData, setPreviewData] = useState([]);

  const { data, errorMsg } = usePost('/api/filter', input);

  function handleChange(value) {
    setInput(value);
  }

  function addItem(item) {
    setPreviewData(previewData.concat(item));
  }

  function removeItem(index) {
    setPreviewData(previewData.filter((item, i) => i !== index));
  }

  return (
    <div className="wrapper">
      <Header previewData={previewData} />
      <div className="container menu-builder">
        <div className="row">
          <div className="col-4">
            <div className="filters">
              <Input data-testid='parent-input' handleChange={(value) => handleChange(value)} />
            </div>
            <ErrorText
              condition={input.length && !data.length}
              text='No results found' />
            <ErrorText
              condition={errorMsg}
              text={errorMsg}
            />
            <List className='item-picker'
              data={data}
              displayButton={false}
              addItem={(item) => addItem(item)} />
          </div>
          <div className="col-8">
            <Title text='Menu preview' />
            <List
              className='menu-preview'
              data={previewData}
              displayButton={true}
              removeItem={(index) => removeItem(index)} />
          </div>
        </div>
      </div>
    </div>
  );
}

Main.propTypes = {
  addItem: PropTypes.func,
  displayButton: PropTypes.bool,
  condition: PropTypes.bool,
  data: PropTypes.array,
  handleClick: PropTypes.func,
  previewData: PropTypes.array,
  removeItem: PropTypes.func,
  text: PropTypes.string,
}
