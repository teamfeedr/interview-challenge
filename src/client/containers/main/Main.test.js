import React from 'react';
import { render, screen } from '@testing-library/react'
import Main from './Main';

describe('Main test', () => {
  it('renders', () => {
    const { container } = render(<Main />)
    expect(container).toMatchSnapshot();
  });
})
