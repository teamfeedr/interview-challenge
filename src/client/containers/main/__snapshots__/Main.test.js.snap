// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main test renders 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="menu-summary"
    >
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-6 menu-summary-left"
          >
            <span>
              0
               items
            </span>
          </div>
          <div
            class="col-6 menu-summary-right"
          />
        </div>
      </div>
    </div>
    <div
      class="container menu-builder"
    >
      <div
        class="row"
      >
        <div
          class="col-4"
        >
          <div
            class="filters"
          >
            <input
              class="form-control"
              data-testid="input-test"
              placeholder="Name"
            />
          </div>
          <ul
            class="item-picker"
            data-testid="list-test"
          />
        </div>
        <div
          class="col-8"
        >
          <h2>
             
            Menu preview
             
          </h2>
          <ul
            class="menu-preview"
            data-testid="list-test"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
